### PR TITLE
Update DeveloperGuide.md

### PR DESCRIPTION
**Pull request by RezwanAhmed123**
The information about the user stories and use cases are not up to date with the planned changes.

This makes the planning phase much more difficult as the information is not consolidated into a single source.

Let's update the developer guide to consolidate all the plans into one place to make it easier for developers to follow the guide while the app is being developed.

This makes the plan more concrete and easy to follow, with less room for ambiguity.

link #20

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -308,32 +308,198 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. User requests to select a client.
+2. System checks if the index is valid.
+3. System selects the client.
+
+Use case ends.
+
+**Extensions**
+
+* 2a. Client index is invalid.
+    * 2a1. System shows invalid index invalid error message.
```
Comment (zi-yii) [inline docs/DeveloperGuide.md:353]: Seems like there is a typo here? It is likely meant to be "2a1. System shows invalid index error message."

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -308,32 +308,198 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  
+    Use case resumes from step 3.
+---
+**Use case 03: Delete a client**
+
+**MSS**
+
+1.  User requests to list clients.
+2.  System shows a list of clients.
+3.  User [selects a client](#UC02) to delete.
```
Comment (zi-yii) [inline docs/DeveloperGuide.md:366]: Nice use of inclusion!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -308,32 +308,198 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Extensions**
+
+* 3a. The client does not have the specified insurance plan.
+    * 3a1. System shows an error message to user.
+
+    Use case ends.
+
+* 3b. The insurance plan id is invalid.
+    * 3b1. System shows an error message to user.
```
Comment (zi-yii) [inline docs/DeveloperGuide.md:396]: This seems to be extension of step 2?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -308,32 +308,198 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2.  System checks if the ID is valid.
+3.  System has selected the insurance plan ID.
 
-*{More to be added}*
+Use case ends.
+
+**Extensions**
+
+* 3a. The client does not have the specified insurance plan.
+    * 3a1. System shows an error message to user.
```
Comment (zi-yii) [inline docs/DeveloperGuide.md:391]: May I know what is this extension here for? I think you may have considered this extension under UC06 instead

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -308,32 +308,198 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+    Use case ends.
+---
+**Use case 06: Remove an insurance plan to a client**
+
+**MSS**
+
+1.  User requests to list clients.
+2.  System shows a list of clients.
+3.  User [selects a client](#UC02), [selects an insurance plan](#UC04) of that client and requests it to be deleted.
```
Comment (zi-yii) [inline docs/DeveloperGuide.md:425]: You are amazing for using inclusions 🤩

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -308,32 +308,198 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 2a. There are no claims for the client.
+    * 2a1. An appropriate message about the client having no claims will be shown.
+
+    Use case ends.
+---
+<a name = "UC08"></a>**Use case 08: Select a claim of a client**
+
+**MSS**
+
+1.  User [selects a client](#UC02), [selects a insurance plan](#UC04) of the client and requests to select a claim of that specific insurance plan.
```
Comment (zi-yii) [inline docs/DeveloperGuide.md:457]: "selects **an** insurance plan" (typo)

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -308,32 +308,198 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 1a. Claim ID is invalid.
+    * 1a1. An appropriate error message will be shown.
+
+    Use case ends.
+---
+**Use case 09: Add a claim to a client**
+
+**MSS**
+
+1.  User [selects a client](#UC02) and [an insurance plan](#UC04) of that client and requests to add a claim to selected insurance plan.
```
Comment (zi-yii) [inline docs/DeveloperGuide.md:473]: Add in "select" so that it is "[select an insurance plan]" (typo)

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -308,32 +308,198 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2.  System adds the claim to the insurance plan.
+
+Use case ends.
+
+**Extensions**
+
+* 3a. The claim already exists for the client.
+
+    * 3a1. System shows an error message.
```
Comment (zi-yii) [inline docs/DeveloperGuide.md:482]: Extension of step 2 instead of step 3

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -308,32 +308,198 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+    Use case ends.
+---
+**Use case 09: Close a claim for a client**
+
+**MSS**
+
+1.  User [selects a claim](#UC08) requests to close that claim for the specified client.
```
Comment (zi-yii) [inline docs/DeveloperGuide.md:490]: Missing "select a client" here, can edit it as:
1. User [selects a client](#UC02), [selects a claim](#UC08) and requests to close that claim for the specified client.

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -347,6 +513,9 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 ### Glossary
 
 * **Mainstream OS**: Windows, Linux, Unix, MacOS
+* **Claims**: A formal request by a client for reimbursement for losses that are covered by specific insurance plans.
+<!-- (the above definition was obtained from: https://www.iciciprulife.com/insurance-claim.html) -->
+* **Claim ID**: A formal claim ID provided by the Insurance Provider when an insurance agent submits a claim on behalf of the client.
```
Comment (zi-yii) [inline docs/DeveloperGuide.md:518]: Maybe we can define what is a valid insurance plan id (insurance plan should exists in the system)

**Review by zi-yii — COMMENTED** [review]
Thank you for the great effort here 🤩 Generally everything else was good, just a few typos to fix and suggestions to consider.

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (matthewyip1511) [reply docs/DeveloperGuide.md:457]: good spot 🙂

**Review by matthewyip1511 — APPROVED** [review]
LGTM, but lets fix the minor changes in another issue 🙂

**Pull request merged**